### PR TITLE
docs: bring up downstream changes found in C8 REST API's generated documentation

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

This PR addresses many small C8 REST API spec issues identified in https://github.com/camunda/camunda-docs/pull/4217.

See that PR for original conversation on these changes. The changes can be categorized as:

* Case consistency
* Shorter path `summary`
* Missing `description`

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -558,14 +558,14 @@ paths:
...
-            The User Task Search successful response.
+            The user task search successful response.
           content:
             application/json:
               schema:
                 $ref: "#/components/schemas/UserTaskSearchQueryResponse"
         "400":
           description: >
-            The User Task Search Query failed.
+            The user task search Query failed.
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:568]: ❓ Should `Query` also be lowercase here? I can see the other query endpoints also use uppercase but I was wondering why that is. In @christinaausley's [original commit](https://github.com/camunda/camunda-docs/pull/4178/commits/45b9e61a03d355cde8de48139f1f1e273db7000a), it was all lowercase.

**Review by tmetzke — APPROVED** [review]
👍 Love it, thanks for feeding this back! 🚀 
❓ There is one question left open from my end otherwise all changes look good.

❓ Understanding question: there are a couple of similar issues still around for other endpoints, especially other search endpoints. Are you addressing those with the original authors or how do we handle them?

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 1 above)
Comment (pepopowitz) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:568]: Yes, this is a mistake on my part! Fixing.

Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:568]:
```suggestion
            The user task search query failed.
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -1245,14 +1245,14 @@ paths:
...
-            The Incident Search successful response.
+            The incident search successful response.
           content:
             application/json:
               schema:
                 $ref: "#/components/schemas/IncidentSearchQueryResponse"
         '400':
           description: >
-            The Incident Search Query failed.
+            The incident search Query failed.
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1255]:
```suggestion
            The incident search query failed.
```

❌ Oops! Another case issue I missed.

**Comment by pepopowitz** [comment]
> ❓ Understanding question: there are a couple of similar issues still around for other endpoints, especially other search endpoints. Are you addressing those with the original authors or how do we handle them?

My intention with this PR was to capture exactly what had been noticed in https://github.com/camunda/camunda-docs/pull/4217, so that I didn't end up down a rabbit hole fixing more issues. 

We should probably do a sweep of the spec for issues that we missed before we were thoroughly reviewing from the DevEx side. I'll bring that to our team meeting next week to talk about when/how we will approach it. What do you think?

**Comment by tmetzke** [comment]
> My intention with this PR was to capture exactly what had been noticed in [camunda/camunda-docs#4217](https://github.com/camunda/camunda-docs/pull/4217), so that I didn't end up down a rabbit hole fixing more issues.
> 
> We should probably do a sweep of the spec for issues that we missed before we were thoroughly reviewing from the DevEx side. I'll bring that to our DevEx team meeting next week to talk about when/how we will approach it. What do you think?

That makes sense, yes, I understand your concern 😅

**Comment by tmetzke** [comment]
It's a GO from my end 👍

**Pull request merged**